pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerBuilderImpl.java

```diff
@@ -228,13 +230,16 @@ private ProducerBuilderImpl(PulsarClientImpl client, ProducerConfigurationData c
     }
 
     @Override
-    public ProducerBuilder<T> property(@NonNull String key, @NonNull String value) {
+    public ProducerBuilder<T> property(String key, String value) {
+        checkArgument(StringUtils.isNotBlank(key), "property key cannot be blank");
         conf.getProperties().put(key, value);
         return this;
     }
 
     @Override
     public ProducerBuilder<T> properties(@NonNull Map<String, String> properties) {
+        properties.keySet().forEach(key ->
```

Comment:
check `properties` is not null before checking keys?

Reply:
yep, it already has `@NonNull` protection. Also, new UT has been added as `testProducerBuilderImplWhenPropertiesIsNull`